docs/arch/scheduler.md

```diff
@@ -0,0 +1,27 @@
+# Scheduler
+
+## What?
+
```

Comment:
The description below assumes that the reader has some knowledge about your overall architecture.

The purpose of this section is to describe, at a high level, what this component is. You should not expect the reader to know what an agent or a step are, or that steps may be ordered somehow.

Make it more generic like e.g.:

```
The `sealci` scheduler receives a stream  of CI jobs units (a.k.a. CI step) as its main input. It also tracks a set of registered computing nodes (a.k.a. CI agent) that provide a dynamic resources pool, where the CI steps will be executed.

The main role of the `sealci` scheduler is to select CI agents where the received CI steps will run and complete. The selection of a CI agent for any given CI step to be executed on is called `scheduling`. 
```

Then you can then elaborate a bit further on the relationship between the agents and the scheduler.